readthedocs/search/api/v3/backend.py

```diff
@@ -0,0 +1,225 @@
+        return list(projects_dict.items())
+
+    def search(self, **kwargs):
+        """
+        Perform the search.
+
+        :param kwargs: All kwargs are passed to the `PageSearch` constructor.
+        """
+        if not self._has_arguments and self.arguments_required:
+            return None
+
+        projects = {project.slug: version.slug for project, version in self.projects}
+        # If the search is done without projects, ES will search on all projects.
+        # If we don't have projects and the user provided arguments,
+        # it means we don't have anything to search on (no results).
+        # Or if we don't have projects and we don't allow searching all,
+        # we also just return.
+        if not projects and (self._has_arguments or not self.default_all):
+            return None
+
+        search = PageSearch(
+            query=self.parser.query,
+            projects=projects,
+            **kwargs,
+        )
+        return search
+
+    def _get_projects_to_search(self):
+        if not self._has_arguments:
+            if self.arguments_required:
+                return None
+            yield from self._get_default_projects()
+            return None
+
+        for value in self.parser.arguments["project"]:
+            project, version = self._get_project_and_version(value)
+            if version and self._has_permission(self.request.user, version):
+                yield project, version
+
+        for value in self.parser.arguments["subprojects"]:
+            project, version = self._get_project_and_version(value)
+
+            # Add the project itself.
+            if version and self._has_permission(self.request.user, version):
+                yield project, version
+
+            # If the user didn't provide a version, version_slug will be `None`,
+            # and we add all subprojects with their default version,
+            # otherwise we will add all projects that match the given version.
+            _, version_slug = self._split_project_and_version(value)
+            if project:
+                yield from self._get_subprojects(
+                    project=project,
+                    version_slug=version_slug,
+                )
+
+        # Add all projects the user has access to.
+        if self.parser.arguments["user"] == "@me":
+            yield from self._get_projects_from_user()
+
+    def _get_projects_from_user(self):
+        for project in Project.objects.for_user(user=self.request.user):
+            version = self._get_project_version(
+                project=project,
+                version_slug=project.default_version,
+                include_hidden=False,
+            )
+            if version and self._has_permission(self.request.user, version):
+                yield project, version
+
+    def _get_subprojects(self, project, version_slug=None):
+        """
+        Get a tuple project/version of all subprojects of `project`.
```

Comment:
```suggestion
        Get a tuple (project,version) of all subprojects of `project`.
```

I thought it was returning a string as used in search, and was confused.